### PR TITLE
Expression parsing and preprocessing

### PR DESCRIPTION
**Pull request by jason0x43**
Handle simple expressions and preprocessor definitions. The `shader.frag` file was updated to include a preprocessor #define and a simple expression just to exercise the parser.

Also do a bit more refactoring:
- Only use a single visitor
- Rename a few more functions and variables for clarity
- Throw some more errors in situations we can't handle properly
- Change the structure of Struct to have a name and an object of fields
- Add more comments. Lots of comments.

### CODE REVIEW

File: glsl-ts-util/src/index.ts [hunk 1]
```diff
@@ -1,34 +1,52 @@
...
 };
 
-interface GlslField {
+// A GLSL const or variable field
+type GlslField = {
   qualifiers: string[];
   identifier: string;
   type: string;
-}
+  initializer?: any;
```
Comment (rorticus) [inline glsl-ts-util/src/index.ts:40]: not an `Initializer` type?

File: glsl-ts-util/src/index.ts [hunk 2]
```diff
@@ -67,215 +92,279 @@ function parseArrayType(str: string) {
...
+      enter: (path: Path<DeclarationStatementNode>) => {
+        // Only process top-level declarations
+        if (
+          path.node.declaration.type !== "declarator_list" ||
+          path.parent?.type !== "program"
+        ) {
+          return;
+        }
+
+        // TODO: structs aren't being parsed
```
Comment (rorticus) [inline glsl-ts-util/src/index.ts:326]: Is this true?

**Review by rorticus — APPROVED** [review]
Nice!

File: glsl-ts-util/src/index.ts (hunk 2 above)
Comment (jason0x43) [reply glsl-ts-util/src/index.ts:326]: D'oh, I forgot to remove that this morning after fixing the struct parsing (that I had broken).

**Pull request merged**